hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/preloader/NearCachePreloader.java

```diff
@@ -107,7 +108,7 @@ public NearCachePreloader(String nearCacheName, NearCachePreloaderConfig preload
 
         String fileName = getFileName(preloaderConfig.getFileName(), nearCacheName);
         this.storeFile = new File(fileName);
-        this.tmpStoreFile = new File(fileName + "~");
+        this.tmpStoreFile = new File(fileName + "-" + newUnsecureUUID());
```

Comment:
It's not easy to pass the real member/client UUID down to the Near Cache pre-loader. So I just use a random UUID. Since it's a temporary file the benefit of having the real UUID seems to be small. If there is an exception there should be the filename in the message anyway.